build/llvm.yml

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+trigger: none
+
+variables:
+  CCACHE_DIR: $(Pipeline.Workspace)/ccache
+  CMAKE_INSTALL_PREFIX: $(Pipeline.Workspace)/LLVM
+  INSTALL_LLVM_PACKAGE: true
+  SCCACHE_DIR: $(Pipeline.Workspace)\sccache
+  AQ_LLVM_PACKAGE_GIT_VERSION: ""
+  ENABLE_COMPILER: false
+  ENABLE_VSIX: false
+  ENABLE_LLVM_BUILDS: true
+  AQ_DOWNLOAD_LLVM: false
+
+jobs:
+- job: GetLlvmTag
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+  - checkout: self
+  - pwsh:  Write-Host "##vso[task.setvariable variable=AQ_LLVM_PACKAGE_GIT_VERSION;isoutput=true]$(Get-Content ./llvm.tag.txt)"
+    workingDirectory: $(System.DefaultWorkingDirectory)/llvm-build
+    displayName: "Get LLVM Tag"
+    name: Tag
+
+- job: LLVMLinuxBuild
+  timeoutInMinutes: 300
+  dependsOn: GetLlvmTag
+  displayName: LLVM Linux Build
+  variables:
+  - name: Aq.Llvm.Package.Git.Version
+    value: $[ dependencies.GetLlvmTag.outputs['Tag.AQ_LLVM_PACKAGE_GIT_VERSION'] ]
+  - name: Pkg.Name
+    value: aq-llvm-x86_64-unknown-linux-gnu-$(Aq.Llvm.Package.Git.Version)
+
+  pool:
+    vmImage: 'ubuntu-20.04'
+
+  steps:
+  - checkout: self
+  - script: |
+      sudo apt-get install -y ccache ninja-build
+      echo "##vso[task.prependpath]/usr/lib/ccache"
+    displayName: Install build dependencies, ccache, and update PATH to use linked versions of gcc, cc, etc
+
+  - task: Cache@2
+    inputs:
+      key: 'version2 | ccache | "$(Agent.OS)"'
+      path: $(CCACHE_DIR)
+      restoreKeys: |
+        version2 | ccache | "$(Agent.OS)"
+    displayName: ccache
+
+  - pwsh: ./llvm-build/llvm.ps1
+    displayName: "LLVM"
+    workingDirectory: $(System.DefaultWorkingDirectory)
+
+  - task: CopyFiles@2
+    inputs:
+      contents: $(System.DefaultWorkingDirectory)/llvm-build/llvm-project/build/$(Pkg.Name).tar.gz
+      targetFolder: $(Build.ArtifactStagingDirectory)
+
+  - task: PublishBuildArtifacts@1
+    inputs:
+      pathToPublish: $(Build.ArtifactStagingDirectory)
+      artifactName: $(Pkg.Name)
+
+
+- job: LLVMWindowsBuild
+  timeoutInMinutes: 300
+  dependsOn: GetLlvmTag
+  displayName: LLVM Windows Build
+  variables:
+  - name: Aq.Llvm.Package.Git.Version
+    value: $[ dependencies.GetLlvmTag.outputs['Tag.AQ_LLVM_PACKAGE_GIT_VERSION'] ]
+  - name: Pkg.Name
+    value: aq-llvm-x86_64-pc-windows-msvc-static-$(Aq.Llvm.Package.Git.Version)
+  pool:
+    vmImage: 'windows-2019'
+
+  steps:
+  - checkout: self
+  - script: |
+      choco install --accept-license -y sccache ninja
+    displayName: Install build dependencies, sccache, and update PATH to use linked versions of gcc, cc, etc
+
+  - task: Cache@2
+    inputs:
+      key: 'version7 | sccache | "$(Agent.OS)"'
+      path: $(SCCACHE_DIR)
+      restoreKeys: |
+        version7 | sccache | "$(Agent.OS)"
+    displayName: sccache
```

Comment:
(I don't know what 'CCACHE', '**S**CCACHE', and 'Cache@2' are in this context)

I see that 
`CCACHE_DIR` is also used outside of `task: Cache@2`, but
`SCCACHE_DIR` is only used in `task: Cache@2`.
That looks suspicious.

Reply:
`Cache@2` is the caching task for Azure DevOps (see https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/cache?view=azure-devops). The environment variables ensure that this tasks loads the cache from the right backend storage.